libs/xcontent/src/main/java/org/elasticsearch/common/xcontent/XContent.java

```diff
@@ -106,4 +108,88 @@ XContentParser createParser(NamedXContentRegistry xContentRegistry,
      */
     XContentParser createParser(NamedXContentRegistry xContentRegistry,
             DeprecationHandler deprecationHandler, Reader reader) throws IOException;
+
+    /**
+     * Low level implementation detail of {@link XContentGenerator#copyCurrentStructure(XContentParser)}.
```

Comment:
Should it be in the `XContentGenerator` class then?

Reply:
That way it can be private. Which just feels better to me.

Reply:
This can't be private (it's used elsewhere here and in x-pack), I can move it if you want but I'm not sure it should go in Generator since it deals with both the Generator and a Parser (which is why I picked XContent)

Reply:
:heart: